contrib/gitian-descriptors/gitian-osx.yml

```diff
@@ -130,6 +131,23 @@ script: |
     make ${MAKEOPTS} -C src check-symbols
     make install-strip DESTDIR=${INSTALLPATH}
 
+    # Hack to work around differences in the behavior of native_cctools' codesign_allocate and Apple distributed codesign and codesign_allocate.
+    # Specifically, the signature is created by appending it to the end of the binary and then updating the size for the section __LINKEDIT
+    # The "vmsize" attribute of this section is rounded to 0x1000, 0x2000, or 0x4000 depending on the architecture. For some reason, this also happens
+    # to differ between what we can build ourselves in native_cctools (which uses Apple's published source code) and what Apple distributes
+    # as "Xcode command line tools". It appears that the Apple distributed tool rounds to 0x2000.
+    #
+    # Fortunately codesign_allocate will not reduce the "vmsize" attribute, and when the signature is appended, the difference in actual size and
+    # "vmsize" will not be present in the final binary.
+    #
+    # Thus we can workaround this by calculating an overestimate of what the signature will be and then using codesign_allocate to preallocate
+    # space for the code signature. Because the same codesign_allocate is used in this step, the rounding problem will not be present.
+    # Since the space allocated for the signature should be way larger than actually needed, the signing step will not reduce "vmsize" so we
+    # won't have any problems there. We use a fixed overestimate because that's easier. Looking at previous releases, the largest MacOS
+    # code signature since 0.12.1 was ~408000 bytes, so using 500000 should be sufficiently large.
```

Comment:
an aside but 500k is crazy large for a cryptographic signature, i only know some of the quantum-hard signature algorithms produce such large signatures, is it known what algorithm they're using?

Reply:
I assume it's just because without --pagesize 0, it splits the binary into pages, and signs every page separately. If a page is 4 kB by default (I don't know what the default is), it's not too surprising to get a huge result.

With --pagesize 0 the signature is just 20 kB. I believe it has still that size because "resources" that are embedded in the binary are still signed separately.

Reply:
I believe it's RSA. The reason the signature is so big is because it's actually multiple signatures. By default, the binary is signed in page sized chunks with each piece getting its own signature. Or at least I think that's what the code is doing.

Reply:
thanks!
yes 20kB isn't that surprising to me for RSA+metadata, and if it signs every page separately that definitely explains things